### PR TITLE
Adds footsteps

### PR DESCRIPTION
**Pull request by nicbn**
[Changelogs]: #
:cl:
soundadd: Nanotrasen shoes no longer contain Silencium. Now footsteps make noise! Sounds from Baystation.
/:cl:

Walking instead of running will reduce the noise, making it now more useful to gameplay.
Plating and wood makes a lot of noise, floor and carpet are quieter.

Range and volume may need to be balanced later on.

Oh, I reorganized some sound files too - the ones beepsky and medbot uses.

https://www.youtube.com/watch?v=TkQdrea9Rx4
https://www.youtube.com/watch?v=Ppt7g9TepSA

### CODE REVIEW

**Comment by MrDoomBringer** [comment]
this better use roblox sounds
upload a video of what it sounds like, maybe with some simplemobs running about too or something

**Comment by nicbn** [comment]
Uploadin' it

**Comment by yoyobatty** [comment]
Shoes and barefeet should sound different. Barefoot on plating is intimidating as fuck.

**Comment by MrDoomBringer** [comment]
i feel like i hear minecraft step sounds somewhere in there
good shit

**Comment by nicbn** [comment]
Clown shoes noise will just stack over the footstep noise

**Comment by MrDoomBringer** [comment]
Wont this still make footsteps if i dont have legs? What if im being dragged on a chair? Do cyborgs make footsteps? How about legless mobs like snakes or blobs?

**Comment by mattroks101** [comment]
It's nice to see features from D2 are finally making their way over to other codebases 6 years later.

**Comment by nicbn** [comment]
Won't play if you are lied or buckled
Cyborgs do make footsteps (should I change this?)
The code excludes all simple animals except some (legless animals won't make footsteps)

**Comment by nicbn** [comment]
I don't think it would be a problem to hear footsteps when being dragged when stunned by Beepsky Smash. I don't know if you'd hear it because I don't know how this stunning works and I'm not bothered enough to test it.

I'm going to randomize the step frequency a little bit.

**Comment by WJohn** [comment]
This whole thing needs a lot more sound effects to work nicely. It's too basic right now. Adding sounds bare feet, jackboots, magboots and related rigsuit sounds, metal clickclacks/rollers/treads/etc for cyborgs and a bunch of other assorted sound effects.

**Comment by Bawhoppen** [comment]
Hurray, this is something we've always needed! 
Now we need to add custom sounds based on your shoe-type, like stomping of jackboots, or toggle-able silence on the syndicate stealth shoes.

**Comment by nicbn** [comment]
I removed the WIP tag because I'll not be adding more features such as barefeet and species-specific different sounds in this PR. The code is pretty simple - there are 118 changed files but most of them are audio files -, so if anyone wants to implement something like that it should be easy.
However, if there is a need for tweaking volume/frequency/range, I'll still do this in the current PR.

**Comment by optimumtact** [comment]
How big are all the sound files?

**Comment by nicbn** [comment]
Around 500 KB

**Comment by praisenarsie** [comment]
i'm sad this isn't even a config. then again i don't know how well it sounds.

**Comment by Doppeltongue** [comment]
Mime shoes should be silent if you didn't already factor that in.

**Comment by SpaceManiac** [comment]
Does not build:
```
code/modules/mob/living/simple_animal/bot/SuperBeepsky.dm:119:error: 'sound/voice/bcriminal.ogg': cannot find file
code/modules/mob/living/simple_animal/bot/SuperBeepsky.dm:119:error: 'sound/voice/bjustice.ogg': cannot find file
code/modules/mob/living/simple_animal/bot/SuperBeepsky.dm:119:error: 'sound/voice/bfreeze.ogg': cannot find file
```

**Comment by bobbahbrown** [comment]
this has frightening implications

**Comment by Bawhoppen** [comment]
These are honestly amazing ingame. Really glad there's a good sound effect for them, otherwise we'd have a ton of people against this.

**Comment by DaxDupont** [comment]
Walking without shoes should make no noise or different noises.
Jackboots should make louder noises

**Comment by epochayur** [comment]
i was pretty impressed by how these turned out in practice

**Comment by nicbn** [comment]
Jackboots could use squeaky component instead of messing with footstep system
For barefeet, robot walk etc.: all of this can be added in the future, but I'm not adding it in this PR

**Comment by nicbn** [comment]
I can move this to a component, should I?

File: code/game/turfs/open.dm [hunk 1]
```diff
@@ -7,17 +7,27 @@
...
+	var/footstep = null
+
 /turf/open/ComponentInitialize()
 	. = ..()
 	if(wet)
 		AddComponent(/datum/component/wet_floor, wet, INFINITY, 0, INFINITY, TRUE)
 	if(LAZYLEN(archdrops))
 		AddComponent(/datum/component/archaeology, archdrops)
 
+/turf/open/Entered(var/mob/living/LM)
```
Comment (vuonojenmustaturska) [inline code/game/turfs/open.dm]: Just to make this point formally, this is not ok. If you want to hook mob movement, use /mob/Moved(). Having the turf check if what entered it is a living mob and all that is nonsense

Comment (nicbn) [reply code/game/turfs/open.dm]: I'll make this into a component

**Comment by cacogen** [comment]
these sound awful where's aranclanos when you need him

**Comment by adamkad1** [comment]
my ears hurt

**Comment by epicmemename** [comment]
While semi interesting this PR leaves out the fact that there is something certainly to be said about sound cues in SS13. Sounds are designed to bring out some sort of reaction, eg, eswords turning on. You can audibly hear that, and it's a required action to turn on an esword, meaning you want to use it at either the last possible second or have it out a while in advance. 

What do footsteps mean? It means that you cannot reliably hide your movement in maint, for example, by avoiding airlocks or going in a cardboard box (which makes no noise iirc). Now every movement telegraphs where you're going to. What's the game effect of this? The fact that you can pinpoint exactly where someone is. The fact that the walk footsteps are quiet does not count, since you can crank the volume up and they're as noisy as airlocks would normally be.

This has huge gameplay implications that remove strategic gameplay relating to being quiet while moving, and instead pushing people to be completely loud (since footsteps are ALWAYS loud), removing one of the most important aspects of SS13, Paranoia. 

If this PR is to have any legitimacy, at the very least, remove the walk footsteps completely. Don't dim.

**Comment by nicbn** [comment]
You can always make another PR to balance a feature
For now walking sound range is smaller than field of view

**Comment by Suicidalpickles** [comment]
Will there be any way to disable footstep sounds via prefs? A bit too repetitive for my liking.

**Comment by nicbn** [comment]
no

**Comment by cacogen** [comment]
i looked for my footstep pr from 1938 with better sounds that a certain nazi killed but i couldn't find it. i hope someone with the patience required to trawl freesound and edit with audacity will replace these abortions of sound with something good

**Comment by Bawhoppen** [comment]
You have impossibly high-standards of sounds if you think these are bad. These are some of the best sounds I've ever heard for ss13.

**Pull request merged**